src/fosslight_scanner/fosslight_scanner.py

```diff
@@ -83,44 +87,44 @@ def set_sub_parameter(default_params, params):  # For dependency
         success = True
+        sheet_list = {}
         if need_init:
             success, final_excel_dir, result_log = init(output_path)
         else:
             final_excel_dir = output_path
         if output_file == "":
-            output_file = _OUTPUT_FILE_PREFIX + _start_time + OUTPUT_FILE_EXTENSION
+            output_file = _OUTPUT_FILE_PREFIX + _start_time
 
         if success:
             output_files = {"SRC": "FL_Source",
                             "BIN": "FL_Binary.xlsx",
                             "DEP": "FL_Dependency.xlsx",
-                            "REUSE": "reuse.xml",
-                            "FINAL": output_file}
+                            "REUSE": "reuse.xml"}
             if src_path != "":
-                call_analysis_api(src_path, "Source Analysis",
-                                  run_scancode.run_scan,
-                                  os.path.abspath(src_path),
-                                  os.path.join(_output_dir, output_files["SRC"]),
-                                  False)
+                result_list = call_analysis_api(src_path, "Source Analysis",
+                                                run_scancode.run_scan,
+                                                os.path.abspath(src_path),
+                                                os.path.join(_output_dir, output_files["SRC"]),
+                                                False, -1, True)
+                sheet_list["SRC_FL_Source"] = [scan_item.get_row_to_print() for scan_item in result_list]
             if dep_path != "":
-                run_analysis(dep_path,
-                             set_sub_parameter(["DEP",
-                                                "-p", os.path.abspath(dep_path),
-                                                "-o", os.path.join(_output_dir, output_files["DEP"])],
-                                               dep_arguments),
-                             dep_main, "Dependency Analysis",
-                             _output_dir, _executed_path)
-
-            ouput_file = os.path.join(final_excel_dir, output_files["FINAL"])
-            success, error_msg = merge_excels(_output_dir, ouput_file)
+                dep_result_list = run_analysis(dep_path, set_sub_parameter(["DEP",
+                                                                            "-p", os.path.abspath(dep_path),
+                                                                            "-o", os.path.join(_output_dir, output_files["DEP"])],
+                                               dep_arguments), dep_main, "Dependency Analysis",
+                                               _output_dir, _executed_path)
+                sheet_list['SRC_FL_Dependency'] = dep_result_list.get('SRC')
+
+            output_file_without_ext = os.path.join(final_excel_dir, output_file)
+            success, msg = write_output_file(output_file_without_ext, OUTPUT_FILE_EXTENSION, sheet_list, scanner='FL-Source')
```

Comment:
@soimkim 
It always generated xlsx file. please modify it (OUTPUT_FILE_EXTENSION).

Reply:
It's my mistake. 😓 I fixed it.